dessia_common/datatools/cluster.py

```diff
@@ -315,12 +315,13 @@ def plot_data(self):
         return Dataset.plot_data(self)
 
     def _plot_data_list(self):
-        _plot_data_list = []
-        for row, label in enumerate(self.labels):
-            _plot_data_list.append({attr: self.matrix[row][col] for col, attr in enumerate(self.common_attributes)})
-            _plot_data_list[-1]["Cluster Label"] = label
-            # (label if label != -1 else "Excluded") plot_data "Excluded" -> NaN
-        return _plot_data_list
+        plot_data_list = []
+        for row, (dobject, label) in enumerate(zip(self.dessia_objects, self.labels)):
+            sample_values = {attr: self.matrix[row][col] for col, attr in enumerate(self.common_attributes)}
+            sample_values["Cluster Label"] = label
+            reference_path = f"dessia_objects/{row}"
+            plot_data_list.append(Sample(sample_values, reference_path, dobject.name))
```

Comment:
I am wondering what would be the input if none of the dessia_objects are named. Maybe we could default it in that case ? For ex : 
```py
name = dessia_object.name
if not name:
    name = f"Sample {row}"
```

Reply:
Done